### PR TITLE
Implemented USDA API w/ Search page

### PR DESCRIPTION
**Pull request by myeken**
- Integrated USDA's food API to provide food macros.
- Created a separate search page for food items.
- Displays food items and their protein, carbs, calories, and fat.

Screenshot 2024-10-07 at 6 30 40 PM

- Fixed an issue using `npm start `command  on MACOS.
- To go to this page, on the search bar after using `npm start` instead of `http://localhost:3000/home`, use `http://localhost:3000/search` to view.

### CODE REVIEW

**Comment by zeshengchen** [comment]
Is this your first pull request? If so, please add me to the list of reviewers.

File: macro-and-me/src/pages/FoodSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    if (query === '') return;
+
+    setLoading(true);
+    setError('');
+
+    try {
+      const apiKey = process.env.REACT_APP_USDA_API_KEY; // USDA API Key from env
```
Comment (zeshengchen) [inline macro-and-me/src/pages/FoodSearch.jsx]: How the env variables will be passed to React app? If through .env file, please install dotenv module and provide ".env_example" file.

**Review by zeshengchen — APPROVED** [review]

**Comment by Nguyen-Noah** [comment]
We should fill the N/A values with 0 in the future to make it more clear but it looks great 👍

**Review by Nguyen-Noah — APPROVED** [review]

**Comment by lattln** [comment]
lgtm

**Review by lattln — APPROVED** [review]
lgtm

**Pull request merged**